geomet/esri.py

```diff
@@ -0,0 +1,202 @@
+    coordinates = data[coordkey]
+    typekey = ([d for d in data if d.lower() == 'type']
+                 or ['type']).pop()
+    if data[typekey].lower() == "polygon":
+        coordinates = [coordinates]
+    part_list = []
+    for part in coordinates:
+        part_item = []
+        for idx, ring in enumerate(part):
+            if idx:
+                part_item.append(None)
+            for coord in ring:
+                part_item.append(coord)
+        if part_item:
+            part_list.append(part_item)
+    return {'rings' : part_list, "spatialReference" : {"wkid" : 4326}}
+#-------------------------------------------------------------------------
+def _dump_esri_point(obj):
+    """
+    Dump a Esri JSON Point to GeoJSON Point.
+
+    :param dict obj:
+        A EsriJSON-like `dict` representing a Point.
+    
+
+    :returns:
+        GeoJSON representation of the Esri JSON Point
+    """
+    if obj.get("x", None) is None or \
+        obj.get("y", None) is None:
+        return {'type': 'Point', 'coordinates': ()}
+    return {'type': 'Point', 'coordinates': (obj.get("x"),
+                                             obj.get("y"))}
+#-------------------------------------------------------------------------
+def _dump_esri_polygon(obj):
+    """
+    Dump a EsriJSON-like Polygon object to GeoJSON.
+
+    Input parameters and return value are the POLYGON equivalent to
+    :func:`_dump_point`.
+    """
+    def split_part(a_part):
+        part_list = []
+        for item in a_part:
+            if item is None:
+                if part_list:
+                    yield part_list
+                part_list = []
+            else:
+                part_list.append((item[0], item[1]))
+        if part_list:
+            yield part_list
+    part_json = [list(split_part(part))
+                    for part in obj['rings']]
+    return {'type': 'MultiPolygon', 'coordinates': part_json}
+#-------------------------------------------------------------------------
+def _dump_esri_multipoint(data):
+    """
+    Dump a EsriJSON-like MultiPoint object to GeoJSON-dict.
+
+    Input parameters and return value are the MULTIPOINT equivalent to
+    :func:`_dump_esri_point`.
+
+    :returns: `dict`
+    """
+
+    return {'type': 'Multipoint', 'coordinates': [pt for pt in data['points']]}
+#-------------------------------------------------------------------------
+def _dump_esri_polyline(data):
+    """
+    Dump a GeoJSON-like MultiLineString object to WKT.
+
+    Input parameters and return value are the MULTILINESTRING equivalent to
+    :func:`_dump_point`.
+    """
+    return {'type': 'MultiLineString', 'coordinates': [[((pt[0], pt[1]) if pt else None)
+                                                        for pt in part]
+                                                        for part in data["paths"]]}
+#-------------------------------------------------------------------------
+_esri_to_geojson_convert = {
```

Comment:
I think you have `load`/`dump` mixed up here. GeoJSON is the reference, so "load" refers to creating GeoJSON from the external format, and "dump" refers to creating the external format from GeoJSON, so for example `_load_geojson_point` should be creating GeoJSON from EsriJSON.

Reply:
@achapkowski, you're using the word "load" in your functions which are, as far as the library is concerned, "dump"ing geometries from GeoJSON to EsriJSON. Because that's confusing from a maintenance perspective, I'd prefer it if you refactored to name them like `_dump_geojson_point`, etc.